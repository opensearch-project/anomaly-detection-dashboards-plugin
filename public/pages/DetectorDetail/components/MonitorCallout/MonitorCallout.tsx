/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';
import { EuiCallOut, EuiLink, EuiIcon } from '@elastic/eui';
import { getAlertingMonitorListLink } from '../../../../utils/utils';

interface MonitorCalloutProps {
  monitorId: string;
  monitorName: string;
}

export const MonitorCallout = (props: MonitorCalloutProps) => {
  return (
    <EuiCallOut
      title="The monitor associated with this detector will not receive any anomaly results"
      color="warning"
      iconType="alert"
    >
      <p>
        Once a detector is stopped, monitor{' '}
        <EuiLink
          href={`${getAlertingMonitorListLink()}/${props.monitorId}`}
          target="_blank"
        >
          {props.monitorName} <EuiIcon type="popout" size="s" />
        </EuiLink>{' '}
        associated with this detector will not receive any anomaly results to
        generate alerts.
      </p>
    </EuiCallOut>
  );
};
