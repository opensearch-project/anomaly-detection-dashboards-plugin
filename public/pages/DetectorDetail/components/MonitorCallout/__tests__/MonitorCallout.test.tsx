/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';
import { render } from '@testing-library/react';
import { MonitorCallout } from '../MonitorCallout';

describe('<MonitorCallout /> spec', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  describe('Monitor callout', () => {
    console.error = jest.fn();
    test('renders component', () => {
      const { container } = render(
        <MonitorCallout
          monitorId="dnexZXEBemUf2kpaW2CK"
          monitorName="test-monitor"
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
