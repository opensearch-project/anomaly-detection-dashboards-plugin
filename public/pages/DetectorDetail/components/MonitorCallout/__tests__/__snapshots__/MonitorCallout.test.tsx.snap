// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MonitorCallout /> spec Monitor callout renders component 1`] = `
<div
  class="euiCallOut euiCallOut--warning"
>
  <div
    class="euiCallOutHeader"
  >
    <svg
      aria-hidden="true"
      class="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiCallOutHeader__icon"
      focusable="false"
      height="16"
      role="img"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
      />
    </svg>
    <span
      class="euiCallOutHeader__title"
    >
      The monitor associated with this detector will not receive any anomaly results
    </span>
  </div>
  <div
    class="euiText euiText--small"
  >
    <div
      class="euiTextColor euiTextColor--default"
    >
      <p>
        Once a detector is stopped, monitor
         
        <a
          class="euiLink euiLink--primary"
          href="/dnexZXEBemUf2kpaW2CK"
          rel="noopener noreferrer"
          target="_blank"
        >
          test-monitor
          <svg
            aria-hidden="true"
            aria-label="External link"
            class="euiIcon euiIcon--small euiIcon-isLoading euiLink__externalIcon"
            focusable="false"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
            />
          </svg>
          <span
            class="euiScreenReaderOnly"
          >
            (opens in a new tab or window)
          </span>
        </a>
         
        associated with this detector will not receive any anomaly results to generate alerts.
      </p>
    </div>
  </div>
</div>
`;
